app/models/document.rb

```diff
@@ -138,6 +138,14 @@ def humanized_document_type
     document_type.underscore.tr("_", " ")
   end
 
+  def update_edition_references!
```

Comment:
nit: should only use a bang method when there is also a non-bang version: https://rubystyle.guide/#dangerous-method-bang